packages/sitecore-jss/src/i18n/graphql.dictionary-service.ts

```diff
@@ -0,0 +1,225 @@
+import { GraphQLRequestClient } from '../graphql-request-client';
+import { SitecoreTemplateId } from '../constants';
+import { DictionaryPhrases, DictionaryServiceBase, CacheOptions } from './dictionary-service';
+
+// TODO: use graphql import instead of string (Anastasiya, March 2021)
+const query = `
+query DictionarySearch(
+    $rootItemId: String!,
+    $language: String!,
+    $dictionaryEntryTemplateId: String!,
+    $pageSize: Int = 10,
+    $after: String
+  ) {
+  search(
+    where: {
+      AND:[
+        { name: "_path",      value: $rootItemId },
+        { name: "_templates", value: $dictionaryEntryTemplateId },
+        { name: "_language",  value: $language }
+      ]
+    }
+    first: $pageSize
+    after: $after
+    orderBy: { name: "Title", direction: ASC }
+  ) {
+    total
+    pageInfo {
+      endCursor
+      hasNext
+    }
+    dictionaryPhrases: results {
+      key: field(name: "key") {
+        value
+      },
+      phrase: field(name: "phrase") {
+        value
+      }
+    }
+  }
+}
+`;
+
+export interface GraphQLDictionaryServiceConfig extends CacheOptions {
+  endpoint: string;
+  rootItemId?: string;
+  pageSize?: number;
+  appName: string;
+}
+
+/**
+ * A reply from the GraphQL Sitecore Dictionary Service
+ */
+type DictionaryQueryResult = {
+  search: {
+    pageInfo: {
+      endCursor: string;
+      hasNext: boolean;
+    };
+    dictionaryPhrases: {
+      key: {
+        value: string;
+      };
+      phrase: {
+        value: string;
+      };
+    }[];
+  };
+};
+
+/**
+ * Fetch dictionary data using the Sitecore Dictionary Service GraphQL API.
+ * Uses graphql-request as the default data fetcher (@see GraphQLRequestClient).
+ */
+export class GraphQLDictionaryService extends DictionaryServiceBase {
+  /**
+   * Creates an instance of graphQL dictionary service with the provided options
+   * @param {GraphQLDictionaryService} options instance
+   */
+  constructor(public options: GraphQLDictionaryServiceConfig) {
+    super(options);
+    this.options.pageSize = this.options.pageSize ?? 10;
```

Comment:
> @anastasiya29 Can you extract `10` into separate variable? May be - `this.DEFAULT_PAGE_SIZE`.
> Like we did in: https://github.com/Sitecore/jss/pull/631/files#diff-65f760d1da619f9554c81966e99c2b15b27ca90226e0f60b53f8dfe3e9927e71R43

Good call. Done.

> Can we bump this default to a higher value to reduce requests? 100??

I chose 10 because the Edge default is 10. Maybe it's better to keep all the page size defaults consistent?

Reply:
@anastasiya29 I can see the consistency point with keeping at 10. I'm fine leaving it at that for now, unless we find that performance is a concern.